Terminal.Gui/Core/View.cs

```diff
@@ -2153,14 +2153,17 @@ bool CanBeVisible (View view)
 		/// </summary>
 		/// <param name="desiredWidth">The desired width.</param>
 		/// <param name="resultWidth">The real result width.</param>
+		/// <param name="currentWidth">The real current width.</param>
 		/// <returns>True if the width can be directly assigned, false otherwise.</returns>
-		public bool SetWidth (int desiredWidth, out int resultWidth)
+		public bool SetWidth (int desiredWidth, out int resultWidth, out int currentWidth)
```

Comment:
It is not a breaking API because it was very recently created by me and I am almost sure that no one uses it yet. It was changed for the better because before it just returned the `resultWidth` based on the `desiredWidth` and in the case of `Button` I needed to know what the current value and that was missing. Naturally we need this for other situations too.

Reply:
Done. @tig sorry for all this disturbance. I hope the solution is your liking. Thanks.